packages/pubsub/src/peer-streams.ts

```diff
@@ -16,6 +16,11 @@ export interface PeerStreamsComponents {
   logger: ComponentLogger
 }
 
+// Define the DecodeOptions type locally
+interface DecoderOptions {
+  maxDataLength?: number
+}
```

Comment:
why not import `DecoderOptions` from `it-length-prefixed`?
```suggestion
import { type DecoderOptions as LpDecoderOptions } from 'it-length-prefixed'

// ...

// Define the DecodeOptions type locally
interface DecoderOptions extends LpDecoderOptions {
  // other custom options we might want for `attachInboundStream`
}
```

Reply:
Yes.
This is because I first tried to modify peer-streams.ts and then I realized I had to change it-length-prefixed to make it work, but then I forgot to import the type, because the code was already written. 
Nice also the `interface DecoderOptions extends LpDecoderOptions ` it would never come to my mind, if you didn't showed me

Reply:
@SgtPooki 
Ops It doesn't wok:

```javascript
Error: @libp2p/pubsub: src/peer-streams.ts(19,15): error TS2305: Module '"it-length-prefixed"' has no exported member 'DecoderOptions'.
```

```typescript
- // Define the DecodeOptions type locally
- interface DecoderOptions {
- maxDataLength?: number
- }

+ import { type DecoderOptions as LpDecoderOptions } from 'it-length-prefixed'
+ // ...
+ // Define the DecodeOptions type locally
+ interface DecoderOptions extends LpDecoderOptions {
+   // other custom options we might want for `attachInboundStream`
+ }
```

Do you want me to modify `it-lenght-prefixed` so that it exports that type to be able to apply this change?

Reply:
I wouldn't worry about changing it again and blocking on it.. but it probably could use 
a tracking issue. You *could* do some typescript fanciness and do something like this:

```
import {decode} from 'it-length-prefixed'

type LpDecoderOptions = NonNullable<Parameters<typeof decode>[1]>
```

which gets the actual type. [See in typescript playground](https://www.typescriptlang.org/play/?ssl=3&ssc=66&pln=1&pc=1#code/JYWwDg9gTgLgBAbwCYFMDGFUF84DMoQhwDkwMAtADYoB2A5jABblhQq7AAeKSxAUHxgBPMCjgAZMABF0mFFADyYGMAg0AznAC8cAHJrdAV0qUAhgCNqAHgAKpqKZAoY89VeGiIuOKgyoAfADaAIwAuv5AA)

Reply:
i'm not 100% sure how @achingbrain feels about this though. He may prefer that type be exported (which is the correct long-term thing to do), but they both result in the same type.

ideally, if `it-length-prefixed` changes parameters, it could break the type here, but that should be a major version bump in `it-length-prefixed` so ultimately the difference is trivial IMO

Reply:
@SgtPooki @achingbrain 
Ok I'm doing this for now thanks
```typescript
import { type decode } from 'it-length-prefixed'
// import { type DecoderOptions as LpDecoderOptions } from 'it-length-prefixed' // Enable this when DecoderOptions will be exported from it-length-prefixed
type LpDecoderOptions = NonNullable<Parameters<typeof decode>[1]>
```
and I will open an issue on `it-length-prefixed` about exporting type and do a PR